wdl/transforms/biscayne/src/main/scala/wdl/transforms/biscayne/linking/expression/values/BiscayneValueEvaluators.scala

```diff
@@ -106,4 +106,49 @@ object BiscayneValueEvaluators {
+    override def evaluateValue(a: Min,
+                               inputs: Map[String, WomValue],
+                               ioFunctionSet: IoFunctionSet,
+                               forCommandInstantiationOptions: Option[ForCommandInstantiationOptions])
+                              (implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[WomValue]] = {
+      val value1 = expressionValueEvaluator.evaluateValue(a.arg1, inputs, ioFunctionSet, forCommandInstantiationOptions)(expressionValueEvaluator)
+      val value2 = expressionValueEvaluator.evaluateValue(a.arg2, inputs, ioFunctionSet, forCommandInstantiationOptions)(expressionValueEvaluator)
+
+      (value1, value2) flatMapN {
+        case (v1, v2) =>
+          val newValue = (v1.value, v2.value) match {
+            case (WomInteger(i1), WomInteger(i2)) => WomInteger(Math.min(i1, i2)).validNel
+            case (WomInteger(i1), WomFloat(l2)) => WomFloat(Math.min(i1.doubleValue, l2)).validNel
+            case (WomFloat(l1), WomInteger(i2)) => WomFloat(Math.min(l1, i2.doubleValue)).validNel
+            case (WomFloat(l1), WomFloat(l2)) => WomFloat(Math.min(l1, l2)).validNel
+            case (other1, other2) => s"Invalid arguments to 'min':(${other1.typeName}, ${other2.typeName})".invalidNel
+          }
+          newValue map { v => EvaluatedValue(v, v1.sideEffectFiles ++ v2.sideEffectFiles) }
+      }
+    }
+  }
+
+  implicit val maxFunctionEvaluator: ValueEvaluator[Max] = new ValueEvaluator[Max] {
+    override def evaluateValue(a: Max,
+                               inputs: Map[String, WomValue],
+                               ioFunctionSet: IoFunctionSet,
+                               forCommandInstantiationOptions: Option[ForCommandInstantiationOptions])(implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[WomValue]] = {
+      val value1 = expressionValueEvaluator.evaluateValue(a.arg1, inputs, ioFunctionSet, forCommandInstantiationOptions)(expressionValueEvaluator)
+      val value2 = expressionValueEvaluator.evaluateValue(a.arg2, inputs, ioFunctionSet, forCommandInstantiationOptions)(expressionValueEvaluator)
+
+      (value1, value2) flatMapN {
+        case (v1, v2) =>
+          val newValue = (v1.value, v2.value) match {
+            case (WomInteger(i1), WomInteger(i2)) => WomInteger(Math.max(i1, i2)).validNel
+            case (WomInteger(i1), WomFloat(l2)) => WomFloat(Math.max(i1.doubleValue, l2)).validNel
+            case (WomFloat(l1), WomInteger(i2)) => WomFloat(Math.max(l1, i2.doubleValue)).validNel
+            case (WomFloat(l1), WomFloat(l2)) => WomFloat(Math.max(l1, l2)).validNel
+            case (other1, other2) => s"Invalid arguments to 'max':(${other1.typeName}, ${other2.typeName})".invalidNel
+          }
+          newValue map { v => EvaluatedValue(v, v1.sideEffectFiles ++ v2.sideEffectFiles) }
```

Comment:
another try to DRY

Reply:
![NO TIME TO DRY](https://user-images.githubusercontent.com/13006282/73696254-ee522680-46a9-11ea-8f50-d05e60788d33.jpg)